### ISSUE TITLE
runtime error message from optimizer.h (CRAN)

### ISSUE THREAD

**Issue by bbolker** [lme4/lme4#561]
```
On 2020-03-20 8:37 a.m., Uwe Ligges wrote:> Thanks, we see:
> 
> lme4.Rcheck/lme4-Ex.Rout:optimizer.h:22:11: runtime error: load of value
> 200, which is not a valid value for type 'bool'
> 
> etc, as for years. Can you please tackle this soon now?
```

This is from [UBSAN](https://www.stats.ox.ac.uk/pub/bdr/memtests/gcc-UBSAN/lme4/lme4-Ex.Rout), and is probably (?) caused by an uninitialized bool somewhere ...

**Comment by bbolker** [comment]
I tried to fix this in 961696f5b8d9 but appear to have broken tests?

**Comment by bbolker** [comment]
I didn't actually break the tests, but I also failed to fix the problem (UBSAN tests with 1.1-23 still showing problems). Would be great to get help from someone knowledgeable with a bit of free time ...

**Comment by bbolker** [comment]
this is fixed now; not sure exactly when, but f839f2f81 is related.
